pdns/dnsbackend.cc

```diff
@@ -230,6 +230,8 @@ bool DNSBackend::getSOA(const DNSName &domain, SOAData &sd)
   int hits=0;
 
   while(this->get(rr)) {
+    if(hits>0)
+      continue;
```

Comment:
should this be after the qtype check?

Reply:
If this is necessary for a specific backed i think you should fix this in the backend.
And put a if(hits>0) throw PDNSException("say something ugly"); here

Reply:
yes, makes sense. I'll change that after #9174 lands